source/grid/grid_in.cc

```diff
@@ -760,6 +760,26 @@ void GridIn<dim, spacedim>::read_ucd (std::istream &in)
 
 
 template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_abaqus (std::istream &in)
+{
+  Assert (tria != 0, ExcNoTriangulationSelected());
+  Assert (dim==2 || dim==3, ExcNotImplemented());
+  AssertThrow (in, ExcIO());
+
+  // Read in the Abaqus file into an intermediate object
+  // that is to be passed along to the UCD reader
+  internal::Abaqus_to_UCD<dim> abaqus_to_ucd;
+  const bool success_abaqus_in = abaqus_to_ucd.read_in_abaqus(in);
+  AssertThrow(success_abaqus_in == true, ExcInternalError());
+  std::stringstream in_ucd;
+  const bool success_ucd_out = abaqus_to_ucd.write_out_avs_ucd(in_ucd);
+  AssertThrow(success_ucd_out == true, ExcInternalError());
+
+  read_ucd(in_ucd);
```

Comment:
Just to reiterate our short discussion last week: I think that using ucd as an intermediate format is a viable and pragmatic approach.